Bugfix: fix typos & params check

Zachariah0318 opened the pull request:
Please refer to the information in commit.

Commit:
1.Fix the spelling error in the DistanceType enum, changing DISTANCE_ENCLIDEAN to DISTANCE_EUCLIDEAN.

Update references to the distance type in related functions.

2.Optimize to_coord and to_rid functions

In Branch 2, only check if the input is of tuple type.

Commit:
Update the coordinate range of the online core

1.Change CORE_Y_ONLINE_MIN from 28 to 20.

2.Change CORE_Y_ONLINE_MAX from 31 to 23.

Commit:
Fix spelling errors

1.Rename PRN_SYN_INTGR_BIT_MAX to PRN_SYN_INTEGR_BIT_MAX (correcting "INTGR" to "INTEGR").

Commit:
Fix disabling logic for max pooling with 1-bit input

Modified the _max_pooling_disable_iw1 function in reg_model.py:

1.Added a condition to check whether max_pooling_en has already been modified.

2.Replaced is with == in the conditional logic (for value comparison instead of identity checking).

Commit:
Update online core base coordinates.

Commit:
Update comments for function gen_magic_init_frame.

Commit:
Fix frame array header check logic

When checking frame array headers, explicitly cast frames[0] to int type. This resolves the right-shift operation issue with np.uint64 type.

Commit:
Restore code formatting.

pre-commit-ci[bot] committed:
:rotating_light: auto fix by pre-commit hooks

pre-commit-ci[bot] committed:
:rotating_light: auto fix by pre-commit hooks

KafCoppelia commented on paicorelib/routing_defs.py:
此处不修改

Zachariah0318 replied:
已回退

Commit:
Revert "Update online core base coordinates."

This reverts commit 9c948496821f1ba00ebf6c3501cef5fd291bd637.

Commit:
Revert "Update online core base coordinates."

This reverts commit 9c948496821f1ba00ebf6c3501cef5fd291bd637.

Commit:
Revert "Restore code formatting."

This reverts commit 7f0c4a1aaec06f6088521c83abc741df3d27f1ad.

Commit:
Revert "Restore code formatting."

This reverts commit 7f0c4a1aaec06f6088521c83abc741df3d27f1ad.

Commit:
Revert "Restore code formatting."

This reverts commit 7f0c4a1aaec06f6088521c83abc741df3d27f1ad.

pre-commit-ci[bot] committed:
:rotating_light: auto fix by pre-commit hooks

Commit:
Merge remote-tracking branch 'PAIlib-Myself/bug-fix' into bug-fix

# Conflicts:
#	paicorelib/routing_defs.py

KafCoppelia commented:
怎么多了这么多“Revert...” commit，不需要将之前的所有修改全部revert哈，这相当于提交了更多的commit

KafCoppelia commented on paicorelib/hw_defs.py:
此处修改后，再顺便fix一下此处的判断：https://github.com/PAICookers/PAIlib/blob/ecf84fbe47874340daddfa08cad8a6eebb461ee8/paicorelib/coordinate.py#L213

对应的测试项目也是错的，需要同步修改：
https://github.com/PAICookers/PAIlib/blob/ecf84fbe47874340daddfa08cad8a6eebb461ee8/tests/test_coordinate.py#L34

Commit:
Fix core type determination logic caused by coordinate range changes

Ensure both x and y coordinates simultaneously satisfy the minimum and maximum limits

Commit:
Update Coordinate and Core Type Test Cases

Change the core type of coordinates (29, 30) and (31, 31) from TYPE_ONLINE to TYPE_OFFLINE.

Add new test cases for coordinates (29, 21), (31, 23), and (28, 22) with core type TYPE_ONLINE.
